my-app/src/pages/newPostPage/AddPostPage.tsx

```diff
@@ -1,77 +1,164 @@
-            </div>
-        </Wrapper>
-    );
-};
-
-const ImageWrapper = styled.img`
-  max-width: 600px;
-  max-height: 600px;
-`
-const TitleWrapper = styled(TextField)`
-  input {
-    font-size: 32px;
-    font-weight: 900;
-  }
-
-  div {
-    &:before,
-    &:after {
-      display: none;
-    }
-  }
-`
-const Wrapper = styled(Paper)`
-  padding: 20px 20px 20px 20px;
-  margin-top: 30px;
-  margin-left: 50px;
-  margin-right: 50px;;
-`
\ No newline at end of file
+import React, {useContext, useRef, KeyboardEvent, useState, useEffect} from 'react'
+import {Button, Paper} from '@mui/material'
+import styled from 'styled-components'
+import {NewPostContext} from './NewPostContext'
+import 'easymde/dist/easymde.min.css'
+import SimpleMdeReact from 'react-simplemde-editor'
+import {useParams} from 'react-router-dom'
+import axios from "../../axios";
+
+
+export const AddPostPage = () => {
+    const {tags,setTags, isEditing, uploadFileHandler,addNewPost, setImageUrl, imageUrl,text, setText, removeFileHandler, title,setTitle} = useContext(NewPostContext)
+    const inputFileRef = useRef<HTMLInputElement | null>(null)
+    const [tag, setTag] = useState<string | null>(null)
+    const {id} = useParams()
+
+    useEffect(()=>{
+        if(id){
+            axios.get(`/posts/${id}`)
+                .then(({data})=>{
+                    setTitle(data.title)
+                    setTags(data.tags)
+                    setText(data.text)
+                    setImageUrl(data.imageUrl)
+                }).catch((err)=>{
+                console.warn(err)
+                alert('Error')
+            })
+        }
+    },[])
+
+
+
+    const tagsHandler = (e: KeyboardEvent<HTMLInputElement>) => {
+        if (e.key === 'Enter') {
+            if (tags.length < 5 && tag?.trim()) {
+                setTags([...tags, tag])
+                setTag(null)
+            }
+        }
+    }
+    const onCloseHandler = (index: number) => {
+        setTags(tags.filter((el, i) => i !== index))
+    }
+
+    return (
+        <Wrapper>
+            <Button
+                onClick={() => inputFileRef?.current?.click()}
+                variant='outlined'
+                size='large'
+            >
+                Upload preview
+            </Button>
+            <input
+                ref={inputFileRef}
+                type='file'
+                onChange={(e) => uploadFileHandler(e)}
+                hidden
+            />
+            {imageUrl && (
+                <>
+                    <Button
+                        variant='contained'
+                        color='error'
+                        onClick={() => removeFileHandler()}
+                    >
+                        Delete
+                    </Button>
+                    <ImageWrapper
+                        src={`http://localhost:3222${imageUrl}`}
+                        alt='Uploaded'
+                    />
+                </>
+            )}
+
+            <br/>
```

Comment:
use css for spacing, <br/> is an indication of bad layout

Reply:
ok, fixed